src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -182,6 +182,11 @@ public void setLevel(int level) {
      * The default compression method will be used if no compression method
      * was specified for the entry, and the current time will be used if
      * the entry has no set modification time.
+     *
+     * The compressed size field will be recalculated for compressed (i.e.
+     * {@code ZipEntry.DEFLATED}) entries when {@link ZipEntry#setCompressedSize(long)}
+     * has not been explicitly called on the {@code ZipEntry}.
+     *
```

Comment:
Here's an alternative that might be a bit clearer to readers.

"When writing a compressed (deflated) entry, and the compressed size has not been explicitly set with the setCompressedSize method, then the compressed size written to the entry's data descriptor will be its actual compressed size."

I think we should put it after the "The default compression method .." sentence and move the sentence on the time stamp to the end.

Reply:
I don't believe we discuss/reference the data descriptor for a Zip entry (outside of the PKWare Zip specification) so I am not sure we should reference it in the javadoc. 

 Placing the sentence after "The default compression method will be used if no compression method was specified for the entry" makes sense

Reply:
Thanks for your input. I've tried to somehow merge both suggestions :)

Did you had a chance to look at the CSR? I think somebody has to review it before I can move it to "Finalized".

Thank you and best regards,
Volker

Reply:
I have added myself as a reviewer based on the updates made by you and Alan